dashboard/test/models/teacher_feedback_test.rb

```diff
@@ -62,4 +62,28 @@ class TeacherFeedbackTest < ActiveSupport::TestCase
     assert_equal(feedbacks[0], TeacherFeedback.where(student: students[0]).latest)
     assert_equal(feedbacks[1], TeacherFeedback.where(student: students[1]).latest)
   end
+
+  test 'destroys when teacher is destroyed' do
+    teacher = create :teacher
+    first_feedback = create :teacher_feedback, teacher: teacher
+    second_feedback = create :teacher_feedback, teacher: teacher
+    teacher.destroy
+    assert_not TeacherFeedback.exists?(first_feedback.id)
+    assert_not TeacherFeedback.exists?(second_feedback.id)
+
+    #Assert that feedback was soft-deleted
+    teacher.undestroy
+    assert TeacherFeedback.exists?(first_feedback.id)
+    assert TeacherFeedback.exists?(second_feedback.id)
```

Comment:
This asserts that the record can be un-deleted. If you just want to verify that it's soft-deleted, you can do something like this:

```ruby
assert first_feedback.reload.deleted?
```
or 
```ruby
assert TeacherFeedback.with_deleted.exists?(first_feedback.id)
```

Reply:
Or just replace the 2 assertions on lines 71 and 72 with something more like my first suggestion here, which will confirm that they are indeed (soft) deleted.